src/apps/js_generic/js_generic_base.cpp

```diff
@@ -436,12 +405,11 @@ namespace ccfapp
               return;
             }
             endpoint_ctx.rpc_ctx->set_response_header(
-              prop_name_cstr, prop_val_cstr);
-            JS_FreeCString(ctx, prop_val_cstr);
+              *prop_name_str, *prop_val_str);
           }
           for (uint32_t i = 0; i < prop_count; i++)
-            JS_FreeAtom(ctx, props[i].atom);
-          js_free(ctx, props);
+            JS_FreeAtom(ctx, prop_enum[i].atom);
```

Comment:
It would be good to have those in an RAII wrapper too, in case an exception gets thrown during the assembly of the header map.